packages/store-sdk/src/storage/useLocalStorage.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Safe localstorage interface. These try..catch are usefull because
+ * some browsers may block accesss to these APIs due to security policies
+ *
+ * Also, the local storage value is lazy-loaded to avoid hydration mismatch
+ * between server/browser. When state is 'hydrated', the in the heap is the
+ * same as the value in local storage
+ */
+import { useState, useEffect, useCallback } from 'react'
+
+const getItem = <T>(key: string) => {
+  try {
+    return JSON.parse(window.localStorage.getItem(key) ?? 'null') as T | null
+  } catch (err) {
+    return null
+  }
+}
+
+const setItem = <T>(key: string, value: T | null) => {
+  try {
+    window.localStorage.setItem(key, JSON.stringify(value))
+  } catch (err) {
+    // noop
+  }
+}
+
+const isFunction = <T>(x: T | (() => T)): x is () => T =>
+  typeof x === 'function'
+
+export const useLocalStorage = <T>(
+  key: string,
+  initialValue: T | (() => T)
+) => {
+  const [data, setData] = useState(() => ({
+    payload: isFunction(initialValue) ? initialValue() : initialValue,
```

Comment:
Maybe pass the key to initialValue? I think it can be useful. Something like 
```suggestion
    payload: isFunction(initialValue) ? initialValue(key) : initialValue,
```

Reply:
why so? if you are passing the key, to this hook, why not use a closure?

Reply:
Oh yeah, you have a point.